_pytest/fixtures.py

```diff
@@ -1037,9 +1037,13 @@ def pytest_generate_tests(self, metafunc):
             if faclist:
                 fixturedef = faclist[-1]
                 if fixturedef.params is not None:
-                    func_params = getattr(getattr(metafunc.function, 'parametrize', None), 'args', [[None]])
+                    parametrize_func = getattr(metafunc.function, 'parametrize', None)
+                    func_params = getattr(parametrize_func, 'args', [[None]])
                     # skip directly parametrized arguments
-                    argnames = func_params[0]
+                    if "argnames" in parametrize_func.kwargs:
```

Comment:
this line directly breaks testing pytest in various cases, see the broken ci results

the defensive coding in the surrounding lines may look ludicrous, but python doesn't have monadic handling of defensive coding so we have to bite the bullet or include null types